### DISCUSSION TITLE
Folia support for QuickShop-Reremake

### DISCUSSION THREAD

**Discussion by virgil698** [PotatoCraft-Studio/QuickShop-Reremake#392] [Ideas]
#Describe the feature
https://github.com/PaperMC/Folia

#Here is a summary of everything made with ChatGPT to save some time:

For plugin developers, there are a few key takeaways from the given text:

Folia introduces regionised multithreading, which means that plugins will require modification to function properly. Plugin developers should expect compatibility with existing plugins to be at zero.

Folia introduces new API additions, such as the RegionisedScheduler and EntityScheduler, which can help plugins run on the correct thread context.

To ensure compatibility with Folia, plugin developers must mark their plugin as compatible by placing "folia-supported: true" in the plugin.yml file. Additionally, plugins must take advantage of the RegionisedScheduler and EntityScheduler API to ensure correct behavior.

Plugins should not assume that they can access or modify data in other regions, as Folia regions tick in parallel and do not share data. Plugin developers should take advantage of the upcoming thread-check API to ensure correct behavior.

There are currently broken API, including the scoreboard API, and planned API changes, such as super aggressive thread checks, which may impact plugin development.

Overall, plugin developers should be prepared for significant changes in their plugins to ensure compatibility with Folia. They should take advantage of the new API additions and mark their plugin as compatible to ensure correct behavior. Additionally, they should be aware of the limitations and rules of Folia's regionised multithreading to avoid bugs and errors in their plugins.

#How is the feature useful to all QuickShop users?
This will be really useful to server owners, as Folia is a game changer but most plugins don't support it yet since it just came out.

#Agreements
 I have searched for and ensured there isn't already an open issue regarding this.

**Comment by YuanYuanOwO** [comment]
https://github.com/PotatoCraft-Studio/QuickShop-Reremake/discussions/382
